### PR TITLE
Updated error returns on rmw_publish to comply with documentation

### PR DESCRIPTION
**Pull request by Lobotuerk**
Signed-off-by: lobotuerk <jtlorente@ekumenlabs.com>

### CODE REVIEW

**Review by clalancette — APPROVED** [review]

File: rmw_fastrtps_shared_cpp/src/rmw_publish.cpp [hunk 1]
```diff
@@ -33,13 +33,14 @@ __rmw_publish(
...
-    ros_message, "ros_message pointer is null", return RMW_RET_ERROR);
+    publisher, "publisher pointer is null", return RMW_RET_INVALID_ARGUMENT);
+  RCUTILS_CHECK_FOR_NULL_WITH_MSG(
+    ros_message, "ros_message pointer is null", return RMW_RET_INVALID_ARGUMENT);
 
   if (publisher->implementation_identifier != identifier) {
     RMW_SET_ERROR_MSG("publisher handle not from this implementation");
-    return RMW_RET_ERROR;
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION;
   }
```
Comment (hidmic) [inline rmw_fastrtps_shared_cpp/src/rmw_publish.cpp:44]: @Lobotuerk consider using `RMW_CHECK_TYPE_IDENTIFIERS_MATCH` instead, like:

```c++
  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
    pub,
    pub->implementation_identifier,
    identifier,
    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
```

Comment (Lobotuerk) [reply rmw_fastrtps_shared_cpp/src/rmw_publish.cpp:44]: ```RMW_CHECK_TYPE_IDENTIFIERS_MATCH``` is not a part of https://github.com/ros2/rmw/blob/master/rmw/include/rmw/error_handling.h, where can I find it?

Comment (Lobotuerk) [reply rmw_fastrtps_shared_cpp/src/rmw_publish.cpp:44]: Nvm, found it

Comment (Lobotuerk) [reply rmw_fastrtps_shared_cpp/src/rmw_publish.cpp]: Changed it to use RMW standard macros

**Review by clalancette — APPROVED** [review]

**Review by Blast545 — APPROVED** [review]

**Review by hidmic — APPROVED** [review]
LGTM pending green CI

**Comment by Lobotuerk** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12271/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7245/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9989/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12162/)

**Pull request merged**